sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -4,18 +4,635 @@
+          WINHTTP_NO_HEADER_INDEX))
+  {
+    GetErrorAndThrow("Error while querying response headers.");
+  }
+
+  // TODO: This check isn't really necessary - need a debug time assert or testing before removing.
+  if (outputBuffer.size() < sizeOfHeaders / sizeof(WCHAR))
+  {
+    GetErrorAndThrow("Unexpected error - buffer size not consistent with expected header size.");
+  }
+
+  auto start = outputBuffer.data();
+  auto last = start + sizeOfHeaders / sizeof(WCHAR);
+  auto statusLineEnd = std::find(start, last, '\0');
+  start = statusLineEnd + 1; // start of headers
+  std::string responseHeaders = WideStringToString(std::wstring(start, last - start));
+
+  DWORD sizeOfHttp = sizeOfHeaders;
+
+  // Get the HTTP version.
+  if (!WinHttpQueryHeaders(
+          handleManager->m_requestHandle,
+          WINHTTP_QUERY_VERSION,
+          WINHTTP_HEADER_NAME_BY_INDEX,
+          outputBuffer.data(),
+          &sizeOfHttp,
+          WINHTTP_NO_HEADER_INDEX))
+  {
+    GetErrorAndThrow("Error while querying response headers.");
+  }
+
+  start = outputBuffer.data();
+  // Assuming ASCII here is OK since the input is expected to be an HTTP version string.
+  std::string httpVersion = WideStringToStringASCII(start, start + sizeOfHttp / sizeof(WCHAR));
+
+  uint16_t majorVersion = 0;
+  uint16_t minorVersion = 0;
+  ParseHttpVersion(httpVersion, &majorVersion, &minorVersion);
+
+  DWORD statusCode = 0;
+  DWORD dwSize = sizeof(statusCode);
+
+  // Get the status code as a number.
+  if (!WinHttpQueryHeaders(
+          handleManager->m_requestHandle,
+          WINHTTP_QUERY_STATUS_CODE | WINHTTP_QUERY_FLAG_NUMBER,
+          WINHTTP_HEADER_NAME_BY_INDEX,
+          &statusCode,
+          &dwSize,
+          WINHTTP_NO_HEADER_INDEX))
+  {
+    GetErrorAndThrow("Error while querying response headers.");
+  }
+
+  HttpStatusCode httpStatusCode = static_cast<HttpStatusCode>(statusCode);
+
+  // Get the optional reason phrase.
+  std::string reasonPhrase;
+  DWORD sizeOfReasonPhrase = sizeOfHeaders;
+
+  if (WinHttpQueryHeaders(
+          handleManager->m_requestHandle,
+          WINHTTP_QUERY_STATUS_TEXT,
+          WINHTTP_HEADER_NAME_BY_INDEX,
+          outputBuffer.data(),
+          &sizeOfReasonPhrase,
+          WINHTTP_NO_HEADER_INDEX))
+  {
+    start = outputBuffer.data();
+    reasonPhrase
+        = WideStringToString(std::wstring(start, start + sizeOfReasonPhrase / sizeof(WCHAR)));
+  }
+
+  // Allocate the instance of the response on the heap with a shared ptr so this memory gets
+  // delegated outside the transport and will be eventually released.
+  auto rawResponse
+      = std::make_unique<RawResponse>(majorVersion, minorVersion, httpStatusCode, reasonPhrase);
+
+  AddHeaders(responseHeaders, rawResponse);
+
+  int64_t contentLength
+      = GetContentLength(handleManager, requestMethod, rawResponse->GetStatusCode());
+
+  rawResponse->SetBodyStream(std::move(GetBodyStream(handleManager, contentLength)));
```

Comment:
GetBodyStream is a function with just one line.
Maybe it is not worth it to have this extra method with just one line.

Reply:
I considered that, but since it is accessing the various handles from the handle manager, I prefer keeping it separate.

Reply:
```suggestion
  rawResponse->SetBodyStream(std::make_unique<WinHttpBodyStream>(std::move(handleManager), contentLength));
```
This way you would be creating a bodyStream which becomes the new owner of the handleManager.